### PR TITLE
Doxygen updates for args_mod, errexit, errmsg, fparsei, fparser

### PR DESCRIPTION
**Pull request by BrianCurtis-NOAA**
Part of #77 

Doxygen updates for:
- args_mod.f
- errexit.f
- errmsg.f
- fparsei.f
- fparser.f

### CODE REVIEW

File: src/args_mod.f [hunk 1]
```diff
@@ -1,12 +1,12 @@
 C> @file
-C
-C> Wrapper for routines iargc and getarg
-C> @author Iredell @date 98-11-DD
+C> @brief Wrapper for routines iargc and getarg.
+C> @author Mark Iredell @date 1998-11-DD
 C>
```
Comment (edwardhartnett) [inline src/args_mod.f]: We need a totally blank line to separate the file-level doxygen block from the module level. So:
```
C> @brief Wrapper for routines iargc and getarg.
C> @author Mark Iredell @date 1998-11-DD

C> This Fortran Module acts as a wrapper to the system
...
```

File: src/errexit.f [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
+C>  - Changed to use XLF utility routine exit_(n) instead of exit(n).
+C>    exit_(n) will return the proper value (n must be 4 byte int)
+C>    to the sh/ksh shell status variable $? ($status for csh)
+C>    on the IBM SP.
 C>
-C> @param[in] IRET INTEGER RETURN CODE
-C>
-C> SUBPROGRAMS CALLED:
-C>   EXIT      - EXITS FROM A FORTRAN PROGRAM
+C> @param[in] IRET Integer return code.
```
Comment (edwardhartnett) [inline src/errexit.f:15]: Surely this is not an in param, but an out param.

File: src/errmsg.f [hunk 3]
```diff
@@ -1,23 +1,18 @@
...
 C>
-C> PROGRAM HISTORY LOG:
-C>   95-10-31  IREDELL
+C> Program history log:
+C>  - 1995-10-31  Mark Iredell
 C>
-C> USAGE:    CALL ERRMSG(CMSG)
-C>   INPUT ARGUMENTS:
-C>     CMSG         CHARACTER*(*) MESSAGE TO WRITE
+C> Usage: call errmsg(cmsg)
```
Comment (edwardhartnett) [inline src/errmsg.f]: Take out all usage lines. Doxygen provides this automatically based on the actual function declaration.

File: src/fparsei.f [hunk 4]
```diff
@@ -1,33 +1,27 @@
...
+C> This subprogram extracts integers from a free-format
+C>  character string. It is useful for parsing command arguments.
 C>
-C> PROGRAM HISTORY LOG:
-C> 1998-09-03  IREDELL  
+C> Program history log:
+C> - 1998-09-03  Mark Iredell
 C>
-C> USAGE:  CALL FPARSEI(CARG,MARG,KARG)
+C> usage: call fparsei(carg, marg, karg)
```
Comment (edwardhartnett) [inline src/fparsei.f]: Remove usage line.

File: src/fparsei.f [hunk 5]
```diff
@@ -1,33 +1,27 @@
...
-C> REMARKS:
-C>   TO DETERMINE THE ACTUAL NUMBER OF INTEGERS FOUND IN THE STRING,
-C>   KARG SHOULD BE SET TO FILL VALUES BEFORE THE CALL TO FPARSEI AND
-C>   THE NUMBER OF NON-FILL VALUES SHOULD BE COUNTED AFTER THE CALL.
-C>
-C> ATTRIBUTES:
-C>   LANGUAGE: FORTRAN 90
+C> @note To determine the actual number of integers found in the string,
+C>   karg should be set to fill values before the call to fparsei() and
+C>   the number of non-fill values should be counted after the call.
```
Comment (edwardhartnett) [inline src/fparsei.f]: Remove the extra spaces at the start of these lines.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: src/errexit.f (hunk 2 above)
Comment (BrianCurtis-NOAA) [reply src/errexit.f:15]: It's weird, if you look at the code it seems to me to be an in param (but also a weird way for an out param). I will move it to an out param if you disagree still.

Comment (edwardhartnett) [reply src/errexit.f:15]: OK, stick with what the code says.

There will be further opportunity to tune the documentation as we write tests. For now, we want to get it all pretty decent, and clear all doxygen warnings, so we can turn on doxygen checking for future PRs. The next step will be to start writing tests, and that will involve updating the documentation some more as we learn about the code.

**Review by edwardhartnett — APPROVED** [review]

File: src/errexit.f (hunk 2 above)
Comment (BrianCurtis-NOAA) [reply src/errexit.f:15]: also noting that the original documentation has it as an input param

**Comment by BrianCurtis-NOAA** [comment]
@edwardhartnett @GeorgeGayno-NOAA @Hang-Lei-NOAA I can't set reviewers yet, can you review please to get this merged?

**Pull request merged**